File: src/main/java/co/fastestpath/api/FastestPathApplication.java
package co.fastestpath.api;

import com.fasterxml.jackson.annotation.JsonAutoDetect.Visibility;
import com.fasterxml.jackson.annotation.PropertyAccessor;
import com.fasterxml.jackson.core.Version;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.SerializationFeature;
import com.fasterxml.jackson.databind.module.SimpleModule;
import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
import com.google.common.annotations.VisibleForTesting;
import com.hubspot.dropwizard.guice.GuiceBundle;
import io.dropwizard.Application;
import io.dropwizard.forms.MultiPartBundle;
import io.dropwizard.setup.Bootstrap;
import io.dropwizard.setup.Environment;

@VisibleForTesting
public class FastestPathApplication extends Application<FastestPathConfiguration> {

  public static void main(String[] args) throws Exception {
    new FastestPathApplication().run(args);
  }

  public GuiceBundle<FastestPathConfiguration> guiceBundle;

  @Override
  public void initialize(Bootstrap<FastestPathConfiguration> bootstrap) {
    guiceBundle = GuiceBundle.<FastestPathConfiguration>newBuilder()
        .setConfigClass(FastestPathConfiguration.class)
        .addModule(new FastestPathModule())
        .enableAutoConfig(getClass().getPackage().getName())
        .build();

    bootstrap.addBundle(new MultiPartBundle());
    bootstrap.addBundle(guiceBundle);
  }

  @Override
  public void run(FastestPathConfiguration configuration, Environment environment) throws Exception {
    ObjectMapper mapper = environment.getObjectMapper();
    mapper.enable(SerializationFeature.INDENT_OUTPUT);
    mapper.disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);
    mapper.setVisibility(PropertyAccessor.FIELD, Visibility.ANY);

    SimpleModule module = new SimpleModule("FastestPathModule", new Version(1, 0, 0, null, null, null));
    mapper.registerModule(module);

    mapper.registerModule(new JavaTimeModule());
  }
}

File: core/src/main/kotlin/co/fastestpath/api/FastestPathApplication.kt
package co.fastestpath.api

import co.fastestpath.api.bootstrap.BootstrapModule
import co.fastestpath.api.bootstrap.Bootstrapper
import co.fastestpath.api.utils.serialization.configureMapper
import com.fasterxml.jackson.core.Version
import com.fasterxml.jackson.databind.module.SimpleModule
import com.hubspot.dropwizard.guice.GuiceBundle
import io.dropwizard.Application
import io.dropwizard.setup.Bootstrap
import io.dropwizard.setup.Environment

open class FastestPathApplication : Application<FastestPathConfiguration>() {

  companion object {
    private val MODULE_VERSION = Version(1, 0, 0, null, null, null)
    private val MODULE = SimpleModule("FastestPathModule", MODULE_VERSION)
  }

  private var bundle : GuiceBundle<FastestPathConfiguration>? = null

  override fun initialize(bootstrap: Bootstrap<FastestPathConfiguration>) {
    bundle = GuiceBundle.newBuilder<FastestPathConfiguration>()
      .setConfigClass(FastestPathConfiguration::class.java)
      .addModule(FastestPathModule())
      .addModule(BootstrapModule())
      .enableAutoConfig(this.javaClass.`package`.name)
      .build()

    bootstrap.addBundle(bundle)
  }

  override fun run(configuration: FastestPathConfiguration, environment: Environment) {
    configureMapper(environment.objectMapper, MODULE)
    val bootstrapper = bundle?.injector?.getInstance(Bootstrapper::class.java)
    bootstrapper?.run()
  }
}
